### PR TITLE
Make repo compliant

### PR DESCRIPTION
**Pull request by RaphaelVogel**
**What this PR does / why we need it**:
* By default all files which do not have license/copyright information in the header are licensed under Apache 2.0 (defined in the `dep5` file). If third-party source code is added/copied into this repo it must contain the third-party copyright and license headers (preferably SPDX headers) and should also be configured in the `dep5` file in section `third-party`.
* Project is registered and checked with the reuse API and a badge is added.
* `LICENSE` file added so that Github recognises the license immediately in the sidebar.

### CODE REVIEW

File: LICENSE.txt [hunk 1]
```diff
@@ -0,0 +1,183 @@
+Apache License
```
Comment (ScheererJ) [inline LICENSE.txt:1]: Please rename the file to `LICENSE` similar to github.com/kubernetes/kubernetes handles it.

**Review by ScheererJ — CHANGES_REQUESTED** [review]

File: LICENSE.txt (hunk 1 above)
Comment (ScheererJ) [reply LICENSE.txt]: Please also update the pull request description accordingly.

File: .reuse/dep5 [hunk 2]
```diff
@@ -6,328 +6,16 @@ Source: https://github.com/gardener/aws-custom-route-controller
...
 Copyright: 2020 SAP SE or an SAP affiliate company and Gardener contributors
 License: Apache-2.0
 
 # --------------------------------------------------
 # documentation
 
-Files:
-  CONTRIBUTING.md
-  README.md
+Files: *.md
```
Comment (MartinWeindel) [inline .reuse/dep5:16]: Does this line have any effect, if you already have `Files: *` above?

**Review by MartinWeindel — COMMENTED** [review]
I have a question about the used file pattern for reuse.

**Review by MartinWeindel — APPROVED** [review]
/lgtm

File: .reuse/dep5 (hunk 2 above)
Comment (RaphaelVogel) [reply .reuse/dep5:16]: yes the dep5 file is processed from top to bottom. The last license assignment wins

File: LICENSE.txt (hunk 1 above)
Comment (RaphaelVogel) [reply LICENSE.txt]: done

**Review by ScheererJ — APPROVED** [review]
/lgtm

**Pull request merged**